test/index.spec.ts

```diff
@@ -334,3 +380,56 @@ test('encryptor:decrypt encrypted data using key derived from wrong password', a
     ),
   ).rejects.toThrow('Incorrect password');
 });
+
+test('encryptor:createKeyFromString generates valid CryptoKey', async ({
+  page,
+}) => {
+  const isKey = await page.evaluate(
+    async (args) => {
+      const key = await window.encryptor.createKeyFromString(
+        args.SAMPLE_EXPORTED_KEY,
+      );
+      return key instanceof CryptoKey;
+    },
+    { SAMPLE_EXPORTED_KEY },
+  );
+  expect(isKey).toBe(true);
+});
+
+test('encryptor:exportKey generates valid CryptoKey string', async ({
+  page,
+}) => {
+  const keyString = await page.evaluate(
+    async (args) => {
+      const key = await window.encryptor.createKeyFromString(
+        args.SAMPLE_EXPORTED_KEY,
+      );
+      return await window.encryptor.exportKey(key);
+    },
+    { SAMPLE_EXPORTED_KEY },
+  );
+  expect(keyString).toStrictEqual(SAMPLE_EXPORTED_KEY);
+});
+
+test('encryptor:encryptWithDetail and decryptWithDetail provide same data ', async ({
```

Comment:
Could you add a test like this, but one that used the key and salt rather than the password to encrypt and decrypt?

Reply:
Maybe these are the additional scenarios to consider:

* Enter a **password** and data, get a vault, key, and salt. Use the **password** to decrypt the vault.
  * i.e. `encryptWithDetail` => `decryptWithDetail`
  * This is the test we have already
* Enter a **password** and data, get a vault, key, and salt. Use the **key** to decrypt the vault.
  * i.e. `encryptWithDetail` => `decryptWithKey`
* Enter a **key** and data, get a vault, key, and salt. Use the **password** to decrypt the vault.
  * i.e. `encryptWithKey` => `decryptWithDetail`
* Enter a **key** and data, get a vault, key, and salt. Use the **key** to decrypt the vault.
  * i.e. `encryptWithKey` => `decryptWithKey`

Reply:
https://github.com/MetaMask/KeyringController/pull/152#discussion_r1012375293: Maybe rethinking the API around `encryptWithDetail`/`decryptWithDetail` can reduce the scenarios (and therefore need for additional unit tests in this module ;))

Reply:
Thank you for the suggestions @Gudahtt !  I've added the requested tests!